lib/constants/theme/system_ui_theme.dart

```diff
@@ -6,85 +6,104 @@ import '../colors.dart';
-    ),
-  ];
-
-  /// Theme where nav bar is [grey] (with default dark theme).
-  /// For light this means [eee].
-  ///
-  /// The opposite is [black].
-  static final grey = [
-    black[0].copyWith(systemNavigationBarColor: AppColors.eee),
-    black[1].copyWith(systemNavigationBarColor: AppColors.grey),
-  ];
+  @override
+  final SystemUiOverlayStyle black;
 
   /// Theme for the drawer screen.
-  static final drawerScreen = [
-    black[0].copyWith(
+  @override
+  final SystemUiOverlayStyle drawerScreen;
+
+  /// Theme for the bottom sheet dialog.
+  @override
+  final SystemUiOverlayStyle bottomSheet;
+
+  /// Theme for the modal dialog.
+  @override
+  final SystemUiOverlayStyle modal;
+
+  @override
+  final Color modalOverGreySystemNavigationBarColor;
+
+  /// Theme for the modal dialog that is displayed over [grey].
+  @override
+  SystemUiOverlayStyle get modalOverGrey => modal.copyWith(
+    systemNavigationBarColor: const Color(0xff0d0d0d),
+  );
+
+  static final SystemUiOverlayStyle _lightBaseStyle = SystemUiOverlayStyle(
+    systemNavigationBarColor: Colors.white,
+    systemNavigationBarDividerColor: Colors.transparent,
+    systemNavigationBarIconBrightness: Brightness.dark,
+    statusBarColor: Colors.white.withOpacity(0.0),
+    statusBarBrightness: Brightness.dark,
+    statusBarIconBrightness: Brightness.dark,
+  );
+
+  static final SystemUiOverlayStyle _darkBaseStyle = SystemUiOverlayStyle(
+    systemNavigationBarColor: Colors.black,
+    systemNavigationBarDividerColor: Colors.transparent,
+    systemNavigationBarIconBrightness: Brightness.light,
+    statusBarColor: AppColors.grey.withOpacity(0.0),
+    statusBarBrightness: Brightness.light,
+    statusBarIconBrightness: Brightness.light,
+  );
+
+  static final SystemUiTheme light = SystemUiTheme(
+    black: _lightBaseStyle,
+    grey: _lightBaseStyle.copyWith(systemNavigationBarColor: AppColors.eee),
+    drawerScreen: _lightBaseStyle.copyWith(
       statusBarColor: Colors.white,
       systemNavigationBarColor: Colors.white,
     ),
-    black[1].copyWith(
-      statusBarColor: AppColors.grey,
-      systemNavigationBarColor: AppColors.grey,
-    ),
-  ];
-
-  /// Theme for the bottom sheet dialog.
-  static final bottomSheet = [
-    black[0].copyWith(
+    bottomSheet: _lightBaseStyle.copyWith(
       systemNavigationBarColor: Colors.white,
       statusBarBrightness: Brightness.light,
       statusBarIconBrightness: Brightness.light,
     ),
-    black[1].copyWith(
-      systemNavigationBarColor: Colors.black,
-    ),
-  ];
-
-  /// Theme for the modal dialog.
-  static final modal = [
-    black[0].copyWith(
+    modal: _lightBaseStyle.copyWith(
       systemNavigationBarColor: const Color(0xff757575),
       systemNavigationBarIconBrightness: Brightness.light,
       statusBarBrightness: Brightness.light,
       statusBarIconBrightness: Brightness.light,
     ),
-    black[1],
-  ];
+    modalOverGreySystemNavigationBarColor: const Color(0xff6d6d6d),
```

Comment:
What is modalOverGreySystemNavigationBarColor and how come it's never used?

Reply:
Ah, sorry about these mistakes. This is a workaround for the problem that all values need to be provided to the constructor when we create the `dark` and `light` constants, and we can't reference a previous argument like we were able to reference previous statics before.

I introduced a getter for `modalOverGray` which is supposed to use the `modalOverGreySystemNavigationBarColor` to construct the `SystemOverlayStyle`. I just forgot to actually use the value, but that is fixed now.

Basically, I turned
```dart
static final modal = [...];
static final modalOverGrey = [
    modal[0].copyWith(systemNavigationBarColor: const Color(0xff6d6d6d)),
    modal[1].copyWith(systemNavigationBarColor: const Color(0xff0d0d0d)),
];
```
into
```dart
class SystemUiTheme {
  final Color modalOverGreySystemNavigationBarColor;
  SystemUiOverlayStyle get modalOverGrey => modal.copyWith(
    systemNavigationBarColor: modalOverGreySystemNavigationBarColor,
  );
}
static final SystemUiTheme light = SystemUiTheme(modalOverGreySystemNavigationBarColor: const Color(0xff6d6d6d));
static final SystemUiTheme dark= SystemUiTheme(modalOverGreySystemNavigationBarColor: const Color(0xff0d0d0d));
```

If I'm missing something obvious to make this better, please let me know.

Reply:
Can we do it like this?
[mypatch.patch](https://github.com/user-attachments/files/15936625/mypatch.patch)

Reply:
That would be nice, but unfortunately no, it produces a `LateInitializationError` when trying to access `SystemUiTheme.light`, because it is not initialized while we construct it.

Reply:
Let's leave it like so then